qt/Utils.h

```diff
@@ -97,6 +99,8 @@ class Utils
             s.startsWith(QStringLiteral("http://")) ||
             s.startsWith(QStringLiteral("https://"));
     }
+
+    static QString getDynamicDownloadDir(Prefs const& prefs, QString const& torrentName);
```

Comment:
And, perhaps the biggest change request -- this is something that would be generically useful in the Mac, Web, Qt, and GTK clients, so it would be _much_ better if this were implemented as much in libtransmission as possible. By adding the dynamic path logic in `qt/`, we're making it harder for other clients to use the feature.